converter/internal/common/common.go

```diff
@@ -106,3 +109,41 @@ func PrettyPrint(in []byte) ([]byte, diag.Diagnostics) {
 	_, _ = buf.WriteString("\n")
 	return buf.Bytes(), nil
 }
+
+// SanitizeRiverIdentifier will sanitize a string to be a valid river identifier.
+// A leading 'id_' will be prepended if the first character is not a letter and any
+// character that is not a letter, number or underscore is replaced with an underscore.
+func SanitizeRiverIdentifier(s string) (string, diag.Diagnostics) {
+	if isValidRiverIdentifier(s) {
+		return s, nil
+	}
+
+	var diags diag.Diagnostics
+	newValue := ""
+
+	for i, c := range s {
+		if i == 0 {
+			if !unicode.IsLetter(c) {
+				newValue = "id_"
+				diags.Add(diag.SeverityLevelWarn, fmt.Sprintf("identifiers must start with a letter. prepended \"id_\" before %q", c))
+			}
+		}
+
+		if !unicode.IsLetter(c) && !unicode.IsNumber(c) && c != '_' {
+			newValue += "_"
+			diags.Add(diag.SeverityLevelWarn, fmt.Sprintf("invalid character %q. replaced with \"+\" since identifiers must only contain letters, numbers, and underscores.", c))
+			continue
+		}
+
+		newValue += string(c)
+	}
+
+	return newValue, diags
```

Comment:
Rather than reimplementing identifier checking, I would instead suggest using River's scanner. See this code here: 

https://github.com/grafana/river/blob/93eb6c45de98ba6b39f5a33690ff82291533f642/parser/internal.go#L711-L715

Reply:
this functionality will be implemented in the river project and used here once complete instead.